raster-foundry/raster-foundry
# Simplify backsplash services with authorization

## Overview

This PR simplifies authorization in backsplash, which allows us to avoid the type gymnastics necessary from an `HTTPService[IO[EitherT[IO, Throwable, ?]]]`, that was the underlying of our previous authorization strategy. That was a drag! This is not a drag.

The main impact is a transition from `authIO <- bigFailureyThing` to an `as user` in the pattern match for reach route.

It also includes changes to ensure we throw specific errors everywhere in preparation for the day when we can upgrade cats-effect, fs2, and cats, and restores implementations in the `MosaicService`, which I think were removed when we hooked up the lab routes for the first time.

Based on what's in #4123, but without upgrades or the typeclass to use the error-handling ADT

### Checklist


## Notes

I left `cats-meow` in `Dependencies` and `Version.scala`, but removed from `backsplash` -- happy to remove from there but didn't think it was hurting anything and wanted a version pin so we didn't have to figure it out again later, since everything was actually working on the branch where it came from except binary compatibility of dependencies.

## Testing Instructions

 * make normal requests that you'd expect to succeed with and without a token and with an invalid token
 * confirm you aren't authorized when you expect not to be
